spec.md

```diff
@@ -146,32 +138,46 @@ that contains both context and data).
   * REQUIRED
   * MUST be a non-empty string
 
-### source
-* Type: Object
-* Description: This describes the software instance that emits the event at
-  runtime (i.e. the producer). It contains sub-properties (listed below)
-* Constraints:
-  * REQUIRED
-  * MUST contain at least one non-empty sub-property.
-
-### source-type
+### source-authority
 * Type: String
-* Description: Type of the event source. Providers define list of event
-  sources.
+* Description: The URI authority component of the source software.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
+  * MUST conform to the
+   ["authority" component](https://tools.ietf.org/html/rfc3986#section-3.2)
+    of the URI spec
 * Examples:
-  * s3
+  * github.com
+  * myenterpriseGitHubInstall.company.com
 
-### source-id
-* Type: String
-* Description: ID of the event source.
+### source-path
+* Type: string
+* Description: The URI path component of the resource that emitted an event.
 * Constraints:
-  * REQUIRED
-  * MUST be a non-empty string
+  * OPTIONAL
+  * If present, MUST conform to the "path-rootless" grammar of the 
+    [URI spec](https://tools.ietf.org/html/rfc3986#section-3.3)
 * Examples:
-  * my.s3.bucket
+  * myorg/myrepo
+  * my/long/ftp/path.jpg
+
+### source-labels
```

Comment:
Do they refer to the "properties" of the source?  
If so, I feel  "source-properties" is a better name since property feels something intrinsic while "label" feels "add-on" . 

Is it TRUE that different event source types could have different source-labels? Are we going to leave the definition of source-labels to event producers?

Reply:
I proposed `source-labels` because that's the use case I had in mind. I'm leaving myself open to being convinced that other systems would be able to add their own labels as well. If we allow this, we should have a discussion about whether they are explicitly annotating the source (letting us keep it `source-labels`) or if they're adding other annotations that shouldn't be tied to the source.

The labels that I'm used to interacting with are under the developer's control, though the developer's software might automatically manage some labels with the developer's credentials. Some examples I've used in the past:

* Google was rolling out some new infrastructure near a major conference. They told speakers to add a specific label on our project if we wanted to defer migration. Because projects (like many resources on Google Cloud) support labels, the team was able to skip over projects that would be used for on-stage demos without any coordination from any other team or a new permanent attribute on their API like "defer_2017_rollout": true
* Firebase releases its own simplified tooling on top of Google Cloud Functions, which provides our customers with a golden road. We want to track when customers use this tooling, so we add a label that says our CLI managed the deploy. Since our CLI experience is K8S style (the customer gives us source and we make the project look like that source), we use this label to avoid deleting functions that are being managed with other tools.

You seem to know dramatically more about IoT and can interpret how the label concept would apply there. It seems totally reasonable to me that the user portal for managing devices would manage labels on those devices. It's also true that there could be a bit of disconnect originally from vendors that they use different label names to refer to similar concepts. Personally, I think this problem will be smaller in practice than deciding that every IoT device can only be correlated by one property.